### PR TITLE
PE-4582: Audio Preview

### PR DESCRIPTION
**Pull request by kunstmusik**
Implements audio media player for public audio files.

--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/4q0odpibmp6i0

### CODE REVIEW

File: lib/pages/drive_detail/components/fs_entry_preview_widget.dart [hunk 1]
```diff
@@ -55,66 +90,716 @@ class VideoPlayerWidget extends StatefulWidget {
...
+    player = AudioPlayer();
+    player.setUrl(widget.audioUrl).then((value) {
+      setState(() {
+        _loadState = LoadState.loaded;
+        _positionListener = player.positionStream.listen((event) {
+          setState(() {});
+        });
+
+        _playStateListener = player.playerStateStream.listen((event) {
+          // logger.d('Player state: $event');
```
Comment (thiagocarvalhodev) [inline lib/pages/drive_detail/components/fs_entry_preview_widget.dart:359]: Commented code

File: lib/pages/drive_detail/components/fs_entry_preview_widget.dart [hunk 2]
```diff
@@ -55,66 +90,716 @@ class VideoPlayerWidget extends StatefulWidget {
...
+        },
+        child: _loadState == LoadState.loading
+            ? const Center(
+                child: SizedBox(
+                  height: 24,
+                  width: 24,
+                  child: CircularProgressIndicator(),
+                ),
+              )
+            // : _loadState == LoadState.failed
```
Comment (thiagocarvalhodev) [inline lib/pages/drive_detail/components/fs_entry_preview_widget.dart:416]: Commented code

File: lib/pages/drive_detail/components/fs_entry_preview_widget.dart [hunk 3]
```diff
@@ -55,66 +90,716 @@ class VideoPlayerWidget extends StatefulWidget {
...
+    var currentTime = getTimeString(videoValue.position);
+    var duration = getTimeString(videoValue.duration);
+
+    return VisibilityDetector(
+        key: const Key('video-player'),
+        onVisibilityChanged: (VisibilityInfo info) {
+          if (mounted) {
+            setState(
+              () {
+                // if (_videoPlayerController.value.isInitialized) {
```
Comment (thiagocarvalhodev) [inline lib/pages/drive_detail/components/fs_entry_preview_widget.dart:180]: Commented code

**Review by thiagocarvalhodev — COMMENTED** [review]
We have a few commented codes. I'll finish the review soon!

File: lib/pages/drive_detail/components/fs_entry_preview_widget.dart (hunk 1 above)
Comment (kunstmusik) [reply lib/pages/drive_detail/components/fs_entry_preview_widget.dart]: Removed

File: lib/pages/drive_detail/components/fs_entry_preview_widget.dart (hunk 2 above)
Comment (kunstmusik) [reply lib/pages/drive_detail/components/fs_entry_preview_widget.dart]: Removed

File: lib/pages/drive_detail/components/fs_entry_preview_widget.dart (hunk 3 above)
Comment (kunstmusik) [reply lib/pages/drive_detail/components/fs_entry_preview_widget.dart:180]: This one is in video player code; will be addressed in PE-4583.

**Review by thiagocarvalhodev — APPROVED** [review]
LGTM!!

**Review by thiagocarvalhodev — APPROVED** [review]

**Pull request merged**